FprimeZephyrReference/Components/ModeManager/ModeManager.fpp

```diff
@@ -0,0 +1,134 @@
+module Components {
+    @ Component to manage system modes and orchestrate safe mode transitions
+    @ based on voltage, watchdog faults, and communication timeouts
+    active component ModeManager {
+
+        # ----------------------------------------------------------------------
+        # Input Ports
+        # ----------------------------------------------------------------------
+
+        @ Port receiving calls from the rate group (1Hz)
+        sync input port run: Svc.Sched
+
+        @ Port to force safe mode entry (callable by other components)
+        async input port forceSafeMode: Fw.Signal
+
+
+        # ----------------------------------------------------------------------
+        # Output Ports
+        # ----------------------------------------------------------------------
+
+        @ Port to notify other components of mode changes
+        output port modeChanged: Fw.Signal
```

Comment:
It looks like a `modeChanged` output port wouldn't be enough for other components to react. I don't think the `Fw.Signal` carries enough information to say what mode the mode manager is in so the down stream components would only know that a mode change occurred. I think you will want either to:
- add an input port so that downstream components can check to see what mode the satellite is in (my vote)
- or you'll want the `modeChanged` output port to return a datatype that carries enough information so downstream components know what mode the satellite has changed into.
- or you'll want who "mode changed" output ports, one for safe mode and another for nominal mode (this is my least favorite implementation)

What are your thoughts? Am I missing something about the implementation?